model/src/pyrenew/model/admissionsmodel.py

```diff
@@ -217,6 +223,19 @@ def sample(
         sample_observed_admissions : For sampling observed hospital admissions
         """
 
+        if n_timepoints is None:
+            if observed_admissions is not None:
+                n_timepoints = len(observed_admissions)
+            else:
+                raise ValueError(
```

Comment:
This is not critical, but can help reduce bugs. Once case: user updates the dataset and forgets to update `n_timepoints` -> Model won't complain and use less data for inference. Perhaps a warning?

Reply:
I'm open to requiring `n_timepoints == observed_admissions.size`; my point was about allowing `n_timepoints` to be inferred from `observed_admissions.size`.

Reply:
Maybe I am not understanding? This is what I don't think is safe:

```python
Model.run(
  n_timepoints = 1000,
  obs = observed_admissions,
)
```

I think this is safer:

```python
Model.run(
  obs = observed_admissions,
)
```

Because `observed_admissions` could change, and the user could "forget" to update `n_timepoints` to the proper number. Now, a different thing is to treat `observed_admissions` as an array, perhaps we could add a test checking for that?

Reply:
> Because observed_admissions could change, and the user could "forget" to update n_timepoints to the proper number. 

Right, but you can have a check and error if `n_timepoints != observed_admissions.size`

> Now, a different thing is to treat observed_admissions as an array, perhaps we could add a test checking for that?
Is it not already an array? Or coerced to one under the hood?

Reply:
We could add a check in the Metaclass, [here](https://github.com/CDCgov/multisignal-epi-inference/blob/b3337c1ca0e8c06027c84bd401c0735fd60c46cf/model/src/pyrenew/metaclass.py#L308-L342).

Reply:
decided in the sync meeting that `HospitalAdmissionsModel.run` will accept either an `n_observed_timepoints_to_simulate` or an `observed_admissions` argument, but not both.

Reply:
Implemented in 8d3efd2